src/daemon/daemon_init_settings.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "daemon_init_settings.h"
+
+#include <multipass/constants.h>
+#include <multipass/platform.h>
+#include <multipass/settings/basic_setting_spec.h>
+#include <multipass/settings/bool_setting_spec.h>
+#include <multipass/settings/dynamic_setting_spec.h>
+#include <multipass/settings/persistent_settings_handler.h>
+#include <multipass/settings/settings.h>
+#include <multipass/utils.h>
+
+#include <QCoreApplication>
+#include <QFileSystemWatcher>
+#include <QObject>
+
+namespace mp = multipass;
+
+namespace
+{
+constexpr const int settings_changed_code = 42;
+
+/*
+ * We make up our own file names to:
+ *   a) avoid unknown org/domain in path;
+ *   b) write daemon config to a central location (rather than user-dependent)
+ * Example: /root/.config/multipassd/multipassd.conf
+ */
+QString persistent_settings_filename()
+{
+    static const auto file_pattern = QStringLiteral("%2%1").arg(mp::settings_extension);
+    static const auto daemon_dir_path = QDir{MP_PLATFORM.daemon_config_home()}; // temporary, replace w/ AppConfigLoc
```

Comment:
Should make sure this is in line with #2485.

Reply:
Do you have anything specific in mind? This was sort of [relocated](https://github.com/canonical/multipass/blob/b5fb2af4bf0ac0e95160d4ee92c3d485e4b3a8f7/src/utils/settings.cpp#L82), so any changes should show up as a merge conflict on either side.